pkg/server/helpers_test.go

```diff
@@ -69,37 +69,37 @@ func TestGetRepoDigestAndTag(t *testing.T) {
 	for desc, test := range map[string]struct {
 		ref                string
 		schema1            bool
-		expectedRepoDigest string
-		expectedRepoTag    string
+		expectedRepoDigest []string
+		expectedRepoTag    []string
 	}{
 		"repo tag should be empty if original ref has no tag": {
 			ref:                "gcr.io/library/busybox@" + digest.String(),
-			expectedRepoDigest: "gcr.io/library/busybox@" + digest.String(),
+			expectedRepoDigest: []string{"gcr.io/library/busybox@" + digest.String()},
 		},
 		"repo tag should not be empty if original ref has tag": {
 			ref:                "gcr.io/library/busybox:latest",
-			expectedRepoDigest: "gcr.io/library/busybox@" + digest.String(),
-			expectedRepoTag:    "gcr.io/library/busybox:latest",
+			expectedRepoDigest: []string{"gcr.io/library/busybox@" + digest.String()},
+			expectedRepoTag:    []string{"gcr.io/library/busybox:latest"},
 		},
 		"repo digest should be empty if original ref is schema1 and has no digest": {
 			ref:                "gcr.io/library/busybox:latest",
 			schema1:            true,
-			expectedRepoDigest: "",
-			expectedRepoTag:    "gcr.io/library/busybox:latest",
+			expectedRepoDigest: []string{},
+			expectedRepoTag:    []string{"gcr.io/library/busybox:latest"},
 		},
 		"repo digest should not be empty if orignal ref is schema1 but has digest": {
 			ref:                "gcr.io/library/busybox@sha256:e6693c20186f837fc393390135d8a598a96a833917917789d63766cab6c59594",
 			schema1:            true,
-			expectedRepoDigest: "gcr.io/library/busybox@sha256:e6693c20186f837fc393390135d8a598a96a833917917789d63766cab6c59594",
-			expectedRepoTag:    "",
+			expectedRepoDigest: []string{"gcr.io/library/busybox@sha256:e6693c20186f837fc393390135d8a598a96a833917917789d63766cab6c59594"},
+			expectedRepoTag:    []string{},
 		},
 	} {
 		t.Logf("TestCase %q", desc)
 		named, err := util.NormalizeImageRef(test.ref)
 		assert.NoError(t, err)
-		repoDigest, repoTag := getRepoDigestAndTag(named, digest, test.schema1)
-		assert.Equal(t, test.expectedRepoDigest, repoDigest)
-		assert.Equal(t, test.expectedRepoTag, repoTag)
+		repoDigest, repoTag := getRepoDigestAndTag(named, named, digest, test.schema1)
+		assert.Equal(t, len(test.expectedRepoDigest), len(repoDigest))
```

Comment:
Only compare the length is not enough...

Reply:
I will update the test case with configs after the design is acceptable.